generators/upgrade/index.js

```diff
@@ -109,24 +132,42 @@ module.exports = UpgradeGenerator.extend({
         detectCurrentBranch: function() {
             var done = this.async();
             this.gitExec(['rev-parse', '-q', '--abbrev-ref', 'HEAD'], function(code, msg, err) {
-                if (code !== 0) this.error('Unable to detect current git branch:\n' + err);
+                if (code !== 0) this.error('Unable to detect current git branch:\n' + msg + ' ' + err);
                 this.sourceBranch = msg.replace('\n','');
                 done();
             }.bind(this));
         },
 
         prepareUpgradeBranch: function() {
             var done = this.async();
-            var createUpgradeBranch = function(callback) {
-                this.gitExec(['branch', '-q', UPGRADE_BRANCH], function(code, msg, err) {
-                    if (code !== 0) this.error('Unable to create ' + UPGRADE_BRANCH + ':\n' + err);
+            var recordCodeHasBeenGenerated = function() {
+                this.gitExec(['merge', '--strategy=ours', '-q', '--no-edit', UPGRADE_BRANCH], function(code, msg, err) {
```

Comment:
that's the whole point of my changes : if `jhipster_upgrade` branch doesn't exist yet, that means that `master` branch holds generated code and eventually code updated/added by the user. So just after creation of the `jhipster_upgrade` branch, in order to avoid loosing user's changes, I create a merge-block commit back on master (no alteration is made on `master` code) ; that indicates to git that `master` branch is up-to-date with `jhipster_upgrade` contents. So, after jhipster version update, next generation (on `jhipster_upgrade` branch) will create a diff, which will be merged on `master` without removing user's changes ! :smile: